### PR TITLE
feat(customs): Rate-limit verification of email codes.

### PR DESCRIPTION
**Pull request by rfk**
This adds basic rate-limiting to attempts to very email codes.  It will need a corresponding customs-server PR to actually support the new actions.  @mozilla/fxa-devs r?

### CODE REVIEW

File: test/mocks.js [hunk 1]
```diff
@@ -126,6 +126,9 @@ function mockDB (data, errors) {
         wrapWrapKb: crypto.randomBytes(32)
       })
     }),
+    forgotPasswordVerified: sinon.spy(function () {
```
Comment (philbooth) [inline test/mocks.js]: You don't actually need to define the body of this function fwiw. `mockObject` provides a default implementation that returns an empty promise, the ones listed here are only to over-ride that with something different.

File: test/local/account_routes.js [hunk 2]
```diff
@@ -1531,16 +1531,17 @@ test('/recovery_email/verify_code', function (t) {
   var mockDB = mocks.mockDB(dbData, dbErrors)
   var mockLog = mocks.spyLog()
   var mockMailer = mocks.mockMailer()
+  var mockCustoms = {
```
Comment (vbudhram) [inline test/local/account_routes.js]: Worthwhile to pull this into [mock.js](https://github.com/mozilla/fxa-auth-server/blob/master/test/mocks.js)?

File: test/local/password_routes.js [hunk 3]
```diff
@@ -84,6 +86,52 @@ test(
...
+      payload: {
+        code: 'abcdef'
+      },
+      query: {}
+    })
+    return new P(function(resolve) {
+      getRoute(passwordRoutes, '/password/forgot/verify_code')
+        .handler(mockRequest, resolve)
+    })
+    .then(function(response) {
```
Comment (vbudhram) [inline test/local/password_routes.js]: Testing locally, response returned is an error `mailer.sendPasswordResetNotification is not a function`. I think using the mock mailer in here will resolve these. Also, check that response has a valid `accountResetToken`.

**Comment by rfk** [comment]
@vbudhram updated based on your feedback, please r?

**Comment by vbudhram** [comment]
@rfk Changes LGTM 👍 , merge at will

**Comment by rfk** [comment]
Thanks!  I'll hold off on merge until https://github.com/mozilla/fxa-customs-server/issues/132 is ready to land as well, in case we decide any changes are needed.

**Pull request merged**